File: sol/solution/0001-0100/0034. Find First and Last Position of Element in Sorted Array/Solution.swift
class Solution {
    func searchRange(_ nums: [Int], _ target: Int) -> [Int] {
        func findBound(_ first: Bool) -> Int {
            var l = 0, r = nums.count - 1, ans = -1
            while l <= r {
                let mid = (l + r) / 2
                if nums[mid] == target {
                    ans = mid
                    if first { r = mid - 1 }
                    else { l = mid + 1 }
                } else if nums[mid] < target {
                    l = mid + 1
                } else {
                    r = mid - 1
                }
            }
            return ans
        }
        return [findBound(true), findBound(false)]
    }
}

File: sol/solution/0001-0100/0034. Find First and Last Position of Element in Sorted Array/Solution.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        return new int[]{findBound(nums, target, true), findBound(nums, target, false)};
    }
    private int findBound(int[] nums, int target, boolean first) {
        int l = 0, r = nums.length - 1, ans = -1;
        while (l <= r) {
            int mid = l + (r - l) / 2;
            if (nums[mid] == target) {
                ans = mid;
                if (first) r = mid - 1;
                else l = mid + 1;
            } else if (nums[mid] < target) l = mid + 1;
            else r = mid - 1;
        }
        return ans;
    }
}